### PR TITLE
Add reschedule reason to DB and emails

### LINKED ISSUE #2709: Replace additional notes for reschedule reason when editing booking
### Is your proposal related to a problem?

When editing booking there should be an input for the user owner that says Reason for reschedule instead of additional comments.

image

### Additional context
Input should be optional, and save to DB field **cancellationReason**.

### PR DESCRIPTION
**Pull request by joeauyeung**
## What does this PR do?

This PR adds the rescheduling reason to the DB and displays it in the emails

Reschedule page
image

Reschedule email
image

Request reschedule email
image

Fixes #2709 

**Environment**: Staging(main branch) / Production

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

## How should this be tested?

- Create a booking
- Reschedule the booking, on the booking page it should look like the above screenshot
- When rescheduling enter a reason
- On the emails, check that the reason is added to the email
- Request to reschedule a booking and add a reason
    - The emails should display the reason

## Checklist

- I haven't added tests that prove my fix is effective or that my feature works
- I haven't checked if new and existing unit tests pass locally with my changes

### CODE REVIEW

File: apps/web/components/booking/pages/BookingPage.tsx [hunk 1]
```diff
@@ -251,6 +252,7 @@ const BookingPage = ({
       email: primaryAttendee.email || "",
       guests: guestListEmails,
       notes: booking.description || "",
+      rescheduleReason: "",
```
Comment (joeauyeung) [inline apps/web/components/booking/pages/BookingPage.tsx:255]: Add reschedule reason to the form

File: apps/web/components/booking/pages/BookingPage.tsx [hunk 2]
```diff
@@ -782,18 +784,31 @@ const BookingPage = ({
                     <label
                       htmlFor="notes"
                       className="mb-1 block text-sm font-medium text-gray-700 dark:text-white">
-                      {t("additional_notes")}
+                      {rescheduleUid ? t("reschedule_optional") : t("additional_notes")}
```
Comment (joeauyeung) [inline apps/web/components/booking/pages/BookingPage.tsx:787]: Change the label if we are creating a new booking or rescheduling one

File: apps/web/components/booking/pages/BookingPage.tsx [hunk 3]
```diff
@@ -782,18 +784,31 @@ const BookingPage = ({
...
+                      <textarea
+                        {...bookingForm.register("notes")}
+                        id="notes"
+                        name="notes"
+                        rows={3}
+                        className={inputClassName}
+                        placeholder={t("share_additional_notes")}
+                        disabled={disabledExceptForOwner}
+                      />
+                    )}
```
Comment (joeauyeung) [inline apps/web/components/booking/pages/BookingPage.tsx:809]: Conditional render if we are creating or rescheduling a booking

File: apps/web/lib/emails/templates/attendee-request-reschedule-email.ts [hunk 4]
```diff
@@ -102,6 +102,7 @@ ${this.calEvent.organizer.language.translate("request_reschedule_title_attendee"
 ${this.calEvent.organizer.language.translate("request_reschedule_subtitle", {
   organizer: this.calEvent.organizer.name,
 })},
+${this.calEvent.rescheduleReason && this.getReason()}
```
Comment (joeauyeung) [inline apps/web/lib/emails/templates/attendee-request-reschedule-email.ts:105]: If there is a rescheduling reason then add it to the email

File: apps/web/lib/emails/templates/attendee-request-reschedule-email.ts [hunk 5]
```diff
@@ -151,6 +152,7 @@ ${getCancelLink(this.calEvent)}
                         <tr>
                           <td align="left" style="font-size:0px;padding:10px 40px;word-break:break-word;">
                             <div style="font-family:Roboto, Helvetica, sans-serif;font-size:16px;font-weight:500;line-height:1;text-align:left;color:#3E3E3E;">
+                              ${this.calEvent.rescheduleReason && this.getReason()}
```
Comment (joeauyeung) [inline apps/web/lib/emails/templates/attendee-request-reschedule-email.ts:155]: If there is a rescheduling reason then add it to the email

File: apps/web/lib/emails/templates/attendee-scheduled-email.ts [hunk 6]
```diff
@@ -452,4 +452,12 @@ ${getRichDescription(this.calEvent)}
...
     return dayjs(this.calEvent.endTime).tz(this.getTimezone());
   }
+
+  protected getReason(): string {
+    return `
+    <div style="line-height: 6px; margin-bottom: 24px;">
+      <p style="color: #494949;">${this.calEvent.attendees[0].language.translate("reschedule_reason")}</p>
+      <p style="color: #494949; font-weight: 400; line-height: 24px;">${this.calEvent.rescheduleReason}</p>
+    </div>`;
+  }
```
Comment (joeauyeung) [inline apps/web/lib/emails/templates/attendee-scheduled-email.ts:462]: Grab the rescheduling reason to add to email

File: apps/web/pages/api/book/event.ts [hunk 7]
```diff
@@ -677,11 +681,18 @@ export default async function handler(req: NextApiRequest, res: NextApiResponse)
 
   if (originalRescheduledBooking?.uid) {
     // Use EventManager to conditionally use all needed integrations.
-    const updateManager = await eventManager.update(evt, originalRescheduledBooking.uid, booking?.id);
+    const updateManager = await eventManager.update(
+      evt,
+      originalRescheduledBooking.uid,
+      booking?.id,
+      reqBody.rescheduleReason
```
Comment (joeauyeung) [inline apps/web/pages/api/book/event.ts:688]: DB transactions happen here so pass the reschedule reason

File: apps/web/pages/api/book/event.ts [hunk 8]
```diff
@@ -712,6 +721,7 @@ export default async function handler(req: NextApiRequest, res: NextApiResponse)
             ...evt,
             additionInformation: metadata,
             additionalNotes, // Resets back to the addtionalNote input and not the overriden value
+            rescheduleReason: reqBody.rescheduleReason,
```
Comment (joeauyeung) [inline apps/web/pages/api/book/event.ts:724]: Pass reschedule reason to email manager

File: packages/core/EventManager.ts [hunk 9]
```diff
@@ -201,6 +202,16 @@ export default class EventManager {
...
+    // Add reschedule reason to new booking
+    await prisma.booking.update({
+      where: {
+        id: newBookingId,
+      },
+      data: {
+        cancellationReason: rescheduleReason,
+      },
+    });
+
```
Comment (joeauyeung) [inline packages/core/EventManager.ts:214]: Add the reschedule reason to the new booking cancellationReason field

**Review by alannnc — COMMENTED** [review]
Updated RescheduleReasons to be handled in the API as cancellationReason.
Updated apps emails accordingly
I'm still validating some emails, will approve once everything looks fine.

**Review by zomars — APPROVED** [review]

**Review by alannnc — APPROVED** [review]

**Pull request merged**